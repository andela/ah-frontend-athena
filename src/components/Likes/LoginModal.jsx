import React from "react";
import { MDBContainer, MDBModal, MDBModalBody, MDBModalHeader } from "mdbreact";
import PropTypes from "prop-types";
import Login from "../../views/LoginView/Login";

const ModalPage = ({ modal, toggle, fallback, history, md }) => {
  return (
    <MDBContainer>
      <MDBModal isOpen={modal} toggle={toggle}>
        <MDBModalHeader toggle={toggle}>
          Please login before you can like or dislike the article
        </MDBModalHeader>
        <MDBModalBody>
          <Login fallback={fallback} history={history} md={md} />
        </MDBModalBody>
      </MDBModal>
    </MDBContainer>
  );
};
ModalPage.propTypes = {
  modal: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
  fallback: PropTypes.string.isRequired,
  history: PropTypes.func.isRequired,
  md: PropTypes.string.isRequired
};
export default ModalPage;
