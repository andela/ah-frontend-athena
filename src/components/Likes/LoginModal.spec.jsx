import React from "react";
import { shallow } from "enzyme";
import LoginModal from "./LoginModal";

let props = {
  modal: "",
  toggle: jest.fn(),
  fallback: "",
  history: jest.fn(),
  md: ""
};
describe("Test if the modal form renders", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LoginModal {...props} />);
  });
  it("Check if the modal renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
