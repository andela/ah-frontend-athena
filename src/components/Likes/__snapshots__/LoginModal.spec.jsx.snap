// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the modal form renders Check if the modal renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalPage
    fallback=""
    history={[MockFunction]}
    md=""
    modal=""
    toggle={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <t
        autoFocus={true}
        backdrop={true}
        backdropTransitionTimeout={150}
        fade={true}
        isOpen=""
        modalTransitionTimeout={300}
        role="dialog"
        tabIndex="-1"
        toggle={[MockFunction]}
        zIndex={1050}
      >
        <t
          closeAriaLabel="Close"
          tag="h4"
          toggle={[MockFunction]}
        >
          Please login before you can like or dislike the article
        </t>
        <t>
          <Connect(Login)
            fallback=""
            history={[MockFunction]}
            md=""
          />
        </t>
      </t>,
      "fluid": false,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "backdrop": true,
        "backdropTransitionTimeout": 150,
        "children": Array [
          <t
            closeAriaLabel="Close"
            tag="h4"
            toggle={[MockFunction]}
          >
            Please login before you can like or dislike the article
          </t>,
          <t>
            <Connect(Login)
              fallback=""
              history={[MockFunction]}
              md=""
            />
          </t>,
        ],
        "fade": true,
        "isOpen": "",
        "modalTransitionTimeout": 300,
        "role": "dialog",
        "tabIndex": "-1",
        "toggle": [MockFunction],
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Please login before you can like or dislike the article",
            "closeAriaLabel": "Close",
            "tag": "h4",
            "toggle": [MockFunction],
          },
          "ref": null,
          "rendered": "Please login before you can like or dislike the article",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(Login)
              fallback=""
              history={[MockFunction]}
              md=""
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "fallback": "",
              "history": [MockFunction],
              "md": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <t
          autoFocus={true}
          backdrop={true}
          backdropTransitionTimeout={150}
          fade={true}
          isOpen=""
          modalTransitionTimeout={300}
          role="dialog"
          tabIndex="-1"
          toggle={[MockFunction]}
          zIndex={1050}
        >
          <t
            closeAriaLabel="Close"
            tag="h4"
            toggle={[MockFunction]}
          >
            Please login before you can like or dislike the article
          </t>
          <t>
            <Connect(Login)
              fallback=""
              history={[MockFunction]}
              md=""
            />
          </t>
        </t>,
        "fluid": false,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "backdrop": true,
          "backdropTransitionTimeout": 150,
          "children": Array [
            <t
              closeAriaLabel="Close"
              tag="h4"
              toggle={[MockFunction]}
            >
              Please login before you can like or dislike the article
            </t>,
            <t>
              <Connect(Login)
                fallback=""
                history={[MockFunction]}
                md=""
              />
            </t>,
          ],
          "fade": true,
          "isOpen": "",
          "modalTransitionTimeout": 300,
          "role": "dialog",
          "tabIndex": "-1",
          "toggle": [MockFunction],
          "zIndex": 1050,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Please login before you can like or dislike the article",
              "closeAriaLabel": "Close",
              "tag": "h4",
              "toggle": [MockFunction],
            },
            "ref": null,
            "rendered": "Please login before you can like or dislike the article",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(Login)
                fallback=""
                history={[MockFunction]}
                md=""
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "fallback": "",
                "history": [MockFunction],
                "md": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
